### PR TITLE
tidied select_canonical_transcripts.pl

### PR DESCRIPTION
**Pull request by dbolser-ebi**
ugly commit I know, but generally tidier.

### CODE REVIEW

**Comment by nerdstrike** [comment]
Is this just perltidy?

**Comment by dbolser-ebi** [comment]
no, that would be simpler / safer I guess.

**Comment by nerdstrike** [comment]
Copyright is now wrong for our EBI-only regime. Why did you change it?
We should probably remove "new Package()" and replace with "Package->new()" for style's sake.
Some of the tidying is welcome, but not everyone needs to stop at 80 characters wide where it consumes vertical space without any logical content.
A second "Done" message with warn?
Subroutines comment is made redundant by the presence of subroutines.

**Comment by dbolser-ebi** [comment]
Agreed.

File: misc-scripts/canonical_transcripts/select_canonical_transcripts.pl [hunk 1]
```diff
@@ -61,183 +65,233 @@
...
     print "Updating database with new canonical transcripts...\n";
     foreach my $change (@change_list) {
-        print "Changin' ".$change->[1]." on ".$change->[0]."\n" if $verbose;
+        print "Changin' ". $change->[1]. " on ". $change->[0]."\n" if $verbose;
         $gene_sth->execute( $change->[1], $change->[0]);
     }
     print "Done\n";
 }
 
+warn "Done\n";
```
Comment (magaliruffier) [inline misc-scripts/canonical_transcripts/select_canonical_transcripts.pl]: a warn for a successful run?

File: misc-scripts/canonical_transcripts/select_canonical_transcripts.pl [hunk 2]
```diff
@@ -61,183 +65,233 @@
...
+                printf $log_fh "New=[%s,%s,%s,%s,%s,%s,'%s']\n",
+                  @{ $transcript_selector->encode_transcript($new_canonical) };
             }
         }
-        
     }
 }
 
-print "Canonical transcript alterations: ".$canonical_changes." from ".$total_genes." genes\n";
-if ($log_fh) {$log_fh->close;}
```
Comment (magaliruffier) [inline misc-scripts/canonical_transcripts/select_canonical_transcripts.pl]: filehandle close seems to have gone missing

File: misc-scripts/canonical_transcripts/select_canonical_transcripts.pl [hunk 3]
```diff
@@ -298,8 +352,10 @@ sub usage {
 
 ./run-healthcheck.sh -d dbname -output problem CoreForeignKeys
 
-A warning about not using CCDS is perfectly acceptible when not running on
-Human, Mouse and Zebrafish.
-";
-    
+A warning about not using CCDS is perfectly acceptible when not
```
Comment (magaliruffier) [inline misc-scripts/canonical_transcripts/select_canonical_transcripts.pl]: accept**a**ble

File: misc-scripts/canonical_transcripts/select_canonical_transcripts.pl (hunk 1 above)
Comment (dbolser-ebi) [inline misc-scripts/canonical_transcripts/select_canonical_transcripts.pl]: Now fixed on this PR (sorry):
https://github.com/Ensembl/ensembl/pull/199/files#diff-03f7b81015cecfc399a731e3d81aa4caR259

File: misc-scripts/canonical_transcripts/select_canonical_transcripts.pl (hunk 2 above)
Comment (dbolser-ebi) [inline misc-scripts/canonical_transcripts/select_canonical_transcripts.pl]: Now fixed on this PR (sorry again):
https://github.com/Ensembl/ensembl/pull/199/files#diff-03f7b81015cecfc399a731e3d81aa4caL198

**Comment by dbolser-ebi** [comment]
No longer touching the licence here:
https://github.com/Ensembl/ensembl/pull/199/files

I suggest you close this one and accept (pending review) that one. Sorry, I totally forgot about this PR,  and found a private commit in my copy of the repo, so submitted it again!


Cheers,
Dan.

**Pull request closed without merging**